start_vscode_stanage.sh

```diff
@@ -182,70 +193,84 @@ fi
 # check that VSC_USERNAME is not an empty string
 if [ -z "$VSC_USERNAME" ]
 then
-        echo -e "Error: No TUoS username is specified, terminating script\n"
+        echo -e "\n Error: No TUoS username is specified, terminating script\n"
         display_help
 else
         echo -e "TUoS username: $VSC_USERNAME"
 fi
 
-# check number of CPU cores
+# check number of CPU per task
 
-# check if VSC_NUM_CPU an integer
-if ! [[ "$VSC_NUM_CPU" =~ ^[0-9]+$ ]]; then
-        echo -e "Error: $VSC_NUM_CPU -> Incorrect format. Please specify number of CPU cores as an integer and try again\n"
+# check if VSC_CPUS_PER_TASK
+if ! [[ "$VSC_CPUS_PER_TASK" =~ ^[0-9c]+$ ]]; then 
+        echo -e "\n Error: $VSC_CPUS_PER_TASK -> Incorrect format. Please specify number of tasks per node as an integer and try again\n"
         display_help
 fi
 
-# check if VSC_NUM_CPU is <= 16
-if [ "$VSC_NUM_CPU" -gt "16" ]; then
-        echo -e "Error: $VSC_NUM_CPU -> Larger than 16. No distributed memory supported, therefore the number of CPU cores needs to be smaller or equal to 16\n"
+# check if VSC_CPUS_PER_TASK is <= 64
+if [ "$VSC_CPUS_PER_TASK" -gt "64" ]; then
+        echo -e "\n Error: $VSC_CPUS_PER_TASK -> Larger than 40. No distributed memory supported, therefore the number of cpus per task needs to be smaller or equal to 40\n"
         display_help
 fi
 
-if [ "$VSC_NUM_CPU" -gt "0" ]; then
-        echo -e "Requesting $VSC_NUM_CPU CPU cores for running the code-server"
+if [ "$VSC_CPUS_PER_TASK" -gt "0" ]; then
+        echo -e "Requesting $VSC_CPUS_PER_TASK cpus per task for running the code-server"
 fi
 
 # check number of GPUs
 
 # check if VSC_NUM_GPU an integer
 if ! [[ "$VSC_NUM_GPU" =~ ^[0-9]+$ ]]; then
-        echo -e "Error: $VSC_NUM_GPU -> Incorrect format. Please specify the number of GPU as an integer and try again\n"
+        echo -e "\n Error: $VSC_NUM_GPU -> Incorrect format. Please specify the number of GPU as an integer and try again\n"
         display_help
 fi
 
-# check if VSC_NUM_GPU is <= 7
-if [ "$VSC_NUM_GPU" -gt "7" ]; then
-        echo -e "Error: No distributed memory supported, therefore number of GPUs needs to be smaller or equal to 7\n"
+# check if VSC_NUM_GPU is <= 4
+if [ "$VSC_NUM_GPU" -gt "4" ]; then
+        echo -e "\n Error: No distributed memory supported, therefore number of GPUs needs to be smaller or equal to 4\n"
         display_help
 fi
```

Comment:
We could do with conditional checks around the value for GPUs given the limits for each GPU type, based on the selected partition where A100s are in ``gpu`` and H100 in ``gpu-h100``.